addons/web/static/src/core/assets.js

```diff
@@ -3,49 +3,52 @@
 import { memoize } from "./utils/functions";
 import { browser } from "./browser/browser";
 import { registry } from "./registry";
+import { session } from "@web/session";
 
-class AssetsLoadingError extends Error {}
+export const assets = {};
 
-/**
- * An object describing a bundle to load
- * @typedef {Object} BundleInfo
- * @property {'script'|'link'} [type] the type of file in this bundle
- * @property {string} [src] the url of the file for this bundle, for this type of file
- * @example `[{"type": "script", "src": "/web/assets/266-d34b0b4/documents_spreadsheet.o_spreadsheet.min.js"}]`
- */
+class AssetsLoadingError extends Error {}
 
 /**
  * Loads the given url inside a script tag.
  *
  * @param {string} url the url of the script
  * @returns {Promise<true>} resolved when the script has been loaded
  */
-export const loadJS = memoize(function loadJS(url) {
+assets.loadJS = memoize(function loadJS(url) {
     if (document.querySelector(`script[src="${url}"]`)) {
         // Already in the DOM and wasn't loaded through this function
         // Unfortunately there is no way to check whether a script has loaded
         // or not (which may not be the case for async/defer scripts)
         // so we assume it is.
         return Promise.resolve();
     }
+
     const scriptEl = document.createElement("script");
     scriptEl.type = "text/javascript";
     scriptEl.src = url;
     document.head.appendChild(scriptEl);
     return new Promise((resolve, reject) => {
-        scriptEl.addEventListener("load", () => resolve(true));
+        scriptEl.addEventListener("load", async () => {
+            const bundle = url.match(/\/web\/assets\/.*\/([^\/]+?)(\.min)?\.js/);
+            if (bundle) {
+                await odoo.ready(bundle[1] + '.bundle.xml');
+            }
```

Comment:
LoadBundle does not have the name, but indeed, only this method will use in bundle and therefore the templates. I will move so that there is this validation only when necessary.